test/gtest/stringify.cpp

```diff
@@ -216,3 +217,26 @@ TEST_F(StringifyTest, Stringify) {
               (uint32_t)-13  // separate block_f if-true
             }));
 }
+
+TEST_F(StringifyTest, Substrings) {
+  Module wasm;
+  parseWast(wasm, dupModuleText);
+
+  HashStringifyWalker stringify = HashStringifyWalker();
+  stringify.walkModule(&wasm);
+
+  SuffixTree st(stringify.hashString);
+  std::vector<SuffixTree::RepeatedSubstring> substrings;
+  for (auto it = st.begin(); it != st.end(); it++) {
+    substrings.push_back(*it);
+  }
+
+  EXPECT_EQ(
+    substrings,
+    (std::vector<SuffixTree::RepeatedSubstring>{
+      SuffixTree::RepeatedSubstring{3u, (std::vector<unsigned>{10, 23})},
+      SuffixTree::RepeatedSubstring{4u, (std::vector<unsigned>{9, 22})},
+      SuffixTree::RepeatedSubstring{2u, (std::vector<unsigned>{11, 24})},
+      SuffixTree::RepeatedSubstring{3u, (std::vector<unsigned>{18, 27})},
+      SuffixTree::RepeatedSubstring{2u, (std::vector<unsigned>{32, 19, 28})}}));
```

Comment:
To verify this, should I be able to see the repeated substrings on line 173-217? If so, maybe add 0,1,2,3 etc. there to make that simpler?